internal/dag/builder_test.go

```diff
@@ -707,7 +707,7 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 					},
 					Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
 						Gateways: &gatewayapi_v1alpha1.RouteGateways{
-							Allow: gatewayAllowTypePtr(gatewayapi_v1alpha1.GatewayAllowAll),
+							Allow: gatewayAllowTypePtr(gatewayapi_v1alpha1.GatewayAllowSameNamespace),
```

Comment:
This was wrong previously and should have been "SameNamespace". The result it still correct, but just fixing up in this PR.

Reply:
Also I think we could DRY this tests up, there's a lot of duplication. 😢